nanofi/src/api/nanofi.cpp

```diff
@@ -457,7 +457,7 @@ int get_content(const flow_file_record* ff, uint8_t* target, int size) {
     return stream->read(target, size);
   } else {
     file_buffer fb = file_to_buffer(ff->contentLocation);
-    size_t copy_size = size < fb.file_len ? size : gsl::narrow<size_t>(fb.file_len);
+    size_t copy_size = static_cast<size_t>(size) < fb.file_len ? size : gsl::narrow<size_t>(fb.file_len);
```

Comment:
I would check for negative `size` before casting to avoid surprises.

Reply:
Added check before cast.